src/main/java/Duke.java

```diff
@@ -1,10 +1,189 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * This class is to build a personal assistance chat-bot called "Kaman"
+ * (Customised from Duke)
+ *
+ * @author NgManSing
+ */
 public class Duke {
+    private static final ArrayList<Task> records = new ArrayList<>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        boolean isLoop = true;
+        printWelcomeMsg();
+
+        do {
+            String rawInput = getUserInput();
+            String[] inputs = rawInput.split(" ");
+            String command = inputs[0];
+            String[] arguments = Arrays.copyOfRange(inputs, 1, inputs.length);
+
+            switch (command) {
+            case "todo":
+                addRecord(arguments, Todo.TASK_TYPE);
+                break;
+            case "deadline":
+                addRecord(arguments, Deadline.TASK_TYPE);
+                break;
+            case "event":
+                addRecord(arguments, Event.TASK_TYPE);
+                break;
+            case "list":
+                showList();
+                break;
+            case "done":
+                executeCommandDone(arguments);
+                break;
+            case "bye":
+                isLoop = executeCommandBye(rawInput, arguments);
+                break;
+            default:
+                promptUserInputInvalid();
+                break;
+            }
```

Comment:
Varying levels of abstraction.

It will be great if you can shift line 19 to 22 and the switch statement to a separate method each, so as to achieve "Single Level of Abstraction Principle".